src/include/hardware.h

```diff
@@ -6,6 +6,16 @@
  * See the file LICENSE for redistribution information.
  */
 
+/*
+ * This macro doesn't do anything and is used for annotation only. We use it to highlight to readers
+ * that the variable is shared across threads, possibly without locking, and requires caution when
```

Comment:
I think the language here can be more precise, as the intent appears to be more precise (or I'm reading too much into it). For example it seems to imply that there is at least one use case where the variable is potentially accessed concurrently by multiple threads outside of lock.

Reply:
Fair point. How about
```
 * This macro doesn't do anything and is used for annotation only. We use it to highlight to readers
 * that the variable is used in lockless inter-thread communication, using mechanisms like memory
 * barriers and compare_and_swap, and requires caution when handling. It is designed to look like a
 * type qualifier.
```
?

Reply:
I think that's great. I'd just suggest "to readers" be removed, as that term is also used liberally to refer to threads that are reading that memory, and I believe "you" (the person reading this) is implied in this context.